modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/delivery/ContentCacheCreator.java

```diff
@@ -45,48 +50,87 @@
     private final int maxChunkSize;
 
     /**
-     * Creates a {@link org.wso2.andes.kernel.distruptor.delivery.ContentCacheCreator} object
+     * Guava based cache used to avoid fetching content for same message id in non-durable topics
+     */
+    private final Cache<Long, DisruptorCachedContent> contentCache;
+
+    /**
+     * Creates a {@link org.wso2.andes.kernel.disruptor.delivery.ContentCacheCreator} object
      * @param maxContentChunkSize maximum content chunk size stored in DB
      */
     public ContentCacheCreator(int maxContentChunkSize) {
         this.maxChunkSize = maxContentChunkSize;
+        contentCache = CacheBuilder.newBuilder().expireAfterWrite(2, TimeUnit.MINUTES).maximumSize(100)
+                                   .concurrencyLevel(1).build();
```

Comment:
Need to decide on the default values for,
- expireAfterWrite
- mazimumSize

Should we make these configurable?